src/Analyzers/CSharp/CodeFixes/UseSystemThreadingLock/CSharpUseSystemThreadingLockFixAllProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeFixesAndRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseSystemThreadingLock;
+
+internal sealed partial class CSharpUseSystemThreadingLockCodeFixProvider
+{
+#if !CODE_STYLE
```

Comment:
❓ What's the issue here? Can we not use DocumentBasedFixAllProvider?

Reply:
> DocumentBasedFixAllProvider

In this case, this is not a DocBasedFixAllProvider (those are for when the fix is only in the doc where the diagnostic is reported).  However, this guy has to fixup references as well.  For example, if at a reference location we did `_gate = new object()` we have to change that to `_gate = new Lock()`.  However, the infrastructure to easily write those types of solution-level fixes, with a SolutionEditor is currently internal to Features layer.  I'm not opposed to someone moving htat into the shared code so that we don't have to do this (we do it in a few places).  I can look into doing that right after this PR.